### ISSUE TITLE
Tabe filterIcon does not work 

### ISSUE THREAD

**Issue by rustam-as** [ant-design/ant-design#44351]
### Reproduction link

[https://codepen.io/rustam-as/pen/qBLELZx?editors=1111](https://codepen.io/rustam-as/pen/qBLELZx?editors=1111)

### Steps to reproduce

created react app with vite and npx
added antd table to my app.jsx
in columns array there is filterIcon for add custom filter and onlick func

### What is expected?

I expect when i click filter icon it consoles some string

### What is actually happening?

it throws error "Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

| Environment | Info |
| --- | --- |
| antd | 5.8.4 |
| React | react@18.2.0 |
| System | Windows 11 pro 64bit (10.0 Build 22000) |
| Browser | Google Chrome Version 110.0.5481.77 (Official Build) (64-bit) |

**Comment by MadCcc** [comment]
https://codesandbox.io/s/shai-xuan-he-pai-xu-antd-5-8-4-forked-mfz35y?file=/demo.tsx
Would this solve your problem?

**Comment by rustam-as** [comment]
@MadCcc no, i need open a modl on filtericon click.

**Comment by NihadMammadli03** [comment]
@MadCcc, good afternoon, the issue is still there, we tried with different versions of "andt" (5.3.2, 5.6.1, 5.8.4) and with different versions of "react". At first we tried it in a project created using yarn/vitejs, however, it is not an issue, because we also tried to use it a CRA and npm projects too. We also used different PC's. The issue still remains, it is impossible to use the filterIcon, as it gives the following error
"Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.". and the suggested solution is not useful, as without filterDropdown we can't change the filter Icon. filterDropdown: true is needed for putting and changing filterIcon.
It is a new bug, in projects that was created weeks ago the filterIcon can be changed.
